File: Numbers/Perfect.cpp
#include<iostream>
using namespace std;

int main()
{
    int iNo = 0,sum=0;
    cout<<"Enter the number : ";
    cin>>iNo;
    
    for(int i=1;i<iNo;i++)
    {
        if((iNo%i) == 0)
        {
            sum = sum + i; // 0+1 , 1+2, 3+3
        }
    }

    if(sum == iNo)
    {
        cout<<"It is a perfect number";
    }
    else
    {
        cout<<"It is not a perfect number";
    }

    return 0;
}

File: Numbers/Perfect.java
import java.util.*;

class Perfect
{
    public static void main(String Arg[])
    {
        int sum=0;

        Scanner sc = new Scanner(System.in);
        System.out.println("Enter a number : ");
        int iNo = sc.nextInt();

        for(int i=1; i<iNo; i++)
        {
            if((iNo % i) == 0)
            {
                sum = sum + i;
            }
        }

        if(sum == iNo)
        {
            System.out.println("It is a perfect number : ");
        }
        else
        {
            System.out.println("It is not a perfect number ");
        }

        sc.close();
    }
}